activerecord/test/cases/relations_test.rb

```diff
@@ -2452,3 +2452,75 @@ def custom_post_relation(alias_name = "omg_posts")
       )
     end
 end
+
+class CreateOrFindByWithinTransactions < ActiveRecord::TestCase
+  unless current_adapter?(:SQLite3Adapter)
+    self.use_transactional_tests = false
+
+    def teardown
+      Subscriber.delete_all
+    end
+
+    def test_multiple_find_or_create_by_within_transactions
+      assert_nil Subscriber.find_by(nick: "bob")
+
+      a_wakeup = Concurrent::Event.new
+      b_wakeup = Concurrent::Event.new
+
+      a = Thread.new do
+        Subscriber.transaction do
```

Comment:
Here the "sqlite3 is busy" exceptions were raised - have no idea why, so made skipping this tests for sqlite. Seems like the same approach is used in https://github.com/rails/rails/blob/744b6715a5d07a2ddffceea8283f7b787541bc60/activerecord/test/cases/transactions_test.rb#L1377 (transactions and threads are also used there)

Reply:
>  have no idea why

It's because sqlite3-ruby doesn't release the GVL.